src/manifest.rs

```diff
@@ -65,262 +65,153 @@ impl From<bare_version::Error> for ManifestParseError {
+            None => "".to_string(),
+        };
+        let metadata = match metadata {
+            Some(metadata) => format!(r#""metadata": {},"#, metadata),
+            None => "".to_string(),
+        };
+        format!(
+            r#"{{
+  "packages": [
+    {{
+      "name": "some",
+      "version": "0.1.0",
+      "id": "some 0.1.0 (path+file:///some)",
+      "manifest_path": "/some/Cargo.toml",
+      {}
+      {}
+      "dependencies": [],
+      "targets": [],
+      "features": {{}},
+      "edition": "2018"
+    }}
+  ],
+  "workspace_members": [
+    "some 0.1.0 (path+file:///some)"
+  ],
+  "target_directory": "/some/target",
+  "version": 1,
+  "workspace_root": "/some"
+}}"#,
+            rust_version, metadata
+        )
     }
 
     #[test]
     fn parse_no_minimum_rust_version() {
-        let contents = r#"[package]
-name = "some"
-version = "0.1.0"
-edition = "2018"
+        let metadata = metadata_json(None, None);
+        let metadata: Metadata = serde_json::from_str(&metadata).unwrap();
 
-[dependencies]
-"#;
-
-        let manifest = CargoManifestParser.parse::<Document>(contents).unwrap();
-
-        let manifest = CargoManifest::try_from(manifest).unwrap();
+        let manifest = CargoManifest::try_from(metadata).unwrap();
 
         assert!(manifest.minimum_rust_version.is_none());
     }
 
     #[test]
     fn parse_rust_version_three_components() {
-        let contents = r#"[package]
-name = "some"
-version = "0.1.0"
-edition = "2018"
-rust-version = "1.56.0"
-
-[dependencies]
-"#;
+        let metadata = metadata_json(Some("1.56.0"), None);
+        let metadata: Metadata = serde_json::from_str(&metadata).unwrap();
 
-        let manifest = CargoManifestParser.parse::<Document>(contents).unwrap();
-
-        let manifest = CargoManifest::try_from(manifest).unwrap();
+        let manifest = CargoManifest::try_from(metadata).unwrap();
         let version = manifest.minimum_rust_version.unwrap();
 
         assert_eq!(version, BareVersion::ThreeComponents(1, 56, 0));
     }
 
     #[test]
     fn parse_rust_version_three_components_with_pre_release() {
-        let contents = r#"[package]
-name = "some"
-version = "0.1.0"
-edition = "2018"
-rust-version = "1.56.0-nightly"
-
-[dependencies]
-"#;
-
-        let manifest = CargoManifestParser.parse::<Document>(contents).unwrap();
-
-        let parse_err = CargoManifest::try_from(manifest).unwrap_err();
-
-        let ManifestParseError { inner } = parse_err;
-        assert_eq!(inner, Error::PreReleaseModifierNotAllowed);
+        let metadata = metadata_json(Some("1.56.0-nightly"), None);
+        // cargo_metadata will check this is invalid for us
+        let err = serde_json::from_str::<Metadata>(&metadata).unwrap_err();
+        assert_eq!(
+            err.to_string(),
+            "pre-release identifiers are not supported in rust-version at line 8 column 38"
+        );
```

Comment:
Tried to keep as much tests as possible, even though this is just confirming cargo metadata's behaviour